app/models/miq_report.rb

```diff
@@ -128,25 +128,13 @@ def self.display_filter_details(cols, mode)
     # mode => :field || :tag
     # Only return cols from has_many sub-tables
     return [] if cols.nil?
-    cols.inject([]) do |r, c|
+    cols.select do |c|
       # eg. c = ["Host.Hardware.Disks : File Name", "Host.hardware.disks-filename"]
-      parts = c.last.split(".")
-      parts[-1] = parts.last.split("-").first # Strip off field name from last element
-
-      if parts.last == "managed"
-        next(r) unless mode == :tag
-        parts.pop # Remove "managed" from tail andjust just evaluate relationship
-      else
-        next(r) unless mode == :field
-      end
-
-      model = parts.shift
-      relats = MiqExpression.get_relats(model)
-      # Build relats hash fetch path like: [:reflections, :hardware, :reflections, :disks, :parent, :multivalue]
-      path = parts.inject([]) { |a, p| a << :reflections; a << p.to_sym }
-      path += [:parent, :multivalue]
-      multi = relats.fetch_path(*path)
-      multi == true ? r << c : r
+      parsed_field = MiqExpression.parse_field_or_tag(c.last)
+      next if !parsed_field.kind_of?(MiqExpression::Tag) && mode == :tag
+      next if !parsed_field.kind_of?(MiqExpression::Field) && mode == :field
```

Comment:
first: I really like the changes you made here.

Not thrilled about the format of `cols` coming in, nor passing in `mode`.
Also keep thinking that these could be tweaked to just call directly into `MiqExpression`

but all in all. see my first line. Good stuff for now.